src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -136,18 +136,19 @@ def launchCluster(self, leaderNodeType, leaderSpotBid, nodeTypes, preemptableNod
             kwargs["subnet_id"] = vpcSubnet
         if not leaderSpotBid:
             logger.info('Launching non-preemptable leader')
-            create_ondemand_instances(ctx.ec2, image_id=self._discoverAMI(ctx),
-                                      spec=kwargs, num_instances=1)
+            instances = create_ondemand_instances(ctx.ec2, image_id=self._discoverAMI(ctx),
+                                                  spec=kwargs, num_instances=1)
+            leader = instances[0]
         else:
             logger.info('Launching preemptable leader')
             # force generator to evaluate
-            list(create_spot_instances(ec2=ctx.ec2,
-                                       price=leaderSpotBid,
-                                       image_id=self._discoverAMI(ctx),
-                                       tags={'clusterName': clusterName},
-                                       spec=kwargs,
-                                       num_instances=1))
-        leader = self._getLeader(clusterName=clusterName, wait=True, zone=zone)
```

Comment:
This waits for the leader. The tests are failing because the container is not yet ready when the test code tries to ssh to the leader.

Reply:
d'oh! thank you! I should be able to fix that while still avoiding another DescribeInstances API call.